Admin UI picard develop access policy tab

SimoneSchmidt opened the pull request:
Added access policy tab to event details dialog

With the merge of this pull request, the access policy tab of the event details dialog will be added. This also includes a component, which can be reused for the access policy tab of the series details dialog.

SimoneSchmidt committed:
Started adding access policy tab to event details dialogue

SimoneSchmidt committed:
added UI-elements for access policy tab

finished adding the UI-elements for the access policy tab of the event details dialog
currently only displays mock-data and no functionality of on-click functions is implemented

SimoneSchmidt committed:
added initial data fetching for event details, policy-tab

added most of the initial data fetching for the policy-tab of the event details dialogue

SimoneSchmidt committed:
finished adding initial data fetching for event details, policy-tab

added the remaining initial data fetching for the policy-tab of the event details dialogue
started implementing the onClick actions (started on New policy on click action)

SimoneSchmidt committed:
finished logic for event details, policy-tab exept for additional actions

-now uses formik for more input fields
-finished adding all thunks
-additional actions still need some work
-automatic removal of obsolete notifications still needed

SimoneSchmidt committed:
Merged current state of admin-ui-picard into admin-ui-picard_develop_access_policy_tab and resolved merge conflicts

merged the current state of admin-ui-picard into admin-ui-picard_develop_access_policy_tab in order to integrate new features and bug fixes
resolved merge conflicts

SimoneSchmidt committed:
Finished Access Policy Tab of Event Details dialog

Finished adding the logic
Changed the form inputs and logic a bit to make better use of formik
Refactored, so that most of the code can be reused for the access policy tab of series details dialog
Cleaned up the code

SimoneSchmidt committed:
Small Fixes to Access Policy Tab of Event Details dialog

Changed a save-button label
Changed a description text to use a more useful version of the description text
Renamed a variable for better readability

isabellakutger commented on modules/admin-ui-picard/app/src/components/shared/modals/ResourceDetailsAccessPolicyTab.js:
```diff
@@ -0,0 +1,457 @@
+/**
+ * This component manages the access policy tab of resource details modals
+ */
+const ResourceDetailsAccessPolicyTab = ({ eventId, header, t, policies, fetchHasActiveTransactions, fetchAccessPolicies, saveNewAccessPolicies,
```
change `eventId` to `resourceId`

isabellakutger commented on modules/admin-ui-picard/app/src/components/shared/modals/ResourceDetailsAccessPolicyTab.js:
```diff
@@ -0,0 +1,457 @@
+
+                                            {/* Save and cancel buttons */}
+                                            { (!transactions.read_only && policyChanged) && (
+                                                <footer style={{padding: '15px'}}>
```
Only show the buttons, if the user actually changed something. Currently, the buttons are still shown after clicking the update button. You can check this using `formik.isValid` and `formik.dirty`.

SimoneSchmidt replied:
This is handled through _policyChanged_. The buttons were only still shown after pressing update, because the update wasn't successful so far (I hadn't added a mock-backend file for this. I have added one now and it should work now). I also forgot to change _policyChanged_ after the update was successful, this was changed now.

isabellakutger commented on modules/admin-ui-picard/app/src/components/shared/modals/ResourceDetailsAccessPolicyTab.js:
```diff
@@ -0,0 +1,457 @@
+                                                        </button>
+                                                    </div>
+                                                    <div className="pull-right">
+                                                        <button onClick={() => saveAccess(formik.values)}
```
Check here, if formik form is changed by the user and deactivate button if not. You can check this by using `formik.isValid ` and ` formik.dirty`.

SimoneSchmidt replied:
If formik form is changed is checked for whole footer. The whole footer with both buttons is only displayed, if the formik form is changed.
I added the _formik.isValid_. Now the button is displayed but disabled, if _!formik.isValid_.

isabellakutger commented on modules/admin-ui-picard/app/src/components/shared/modals/ResourceDetailsAccessPolicyTab.js:
```diff
@@ -0,0 +1,457 @@
+ */
+const ResourceDetailsAccessPolicyTab = ({ eventId, header, t, policies, fetchHasActiveTransactions, fetchAccessPolicies, saveNewAccessPolicies,
+                                          addNotification, fetchAclTemplates, fetchRoles}) => {
+
```
Please add some comments to this file

isabellakutger commented on modules/admin-ui-picard/app/src/reducers/eventDeatilsReducers.js:
```diff
@@ -18,10 +20,46 @@ const initialState = {
+        case LOAD_EVENT_POLICIES_SUCCESS: {
+            const { accessPolicies } = payload;
+            let policies = [];
+            if(!!accessPolicies.episode_access){
```
Please move this code to the thunk where you fetching the episodes access policies. The redux state should only contain the actual state changes.

isabellakutger commented on modules/admin-ui-picard/app/src/thunks/eventDetailsThunks.js:
```diff
@@ -19,6 +21,54 @@ const getHttpHeaders = () => {
+}
+
+export const fetchAccessPolicies = (eventId) => async (dispatch) => {
+    try {
```
add an action for  "load event polices in process"

isabellakutger commented on modules/admin-ui-picard/app/src/thunks/eventDetailsThunks.js:
```diff
@@ -19,6 +21,54 @@ const getHttpHeaders = () => {
+
+        dispatch(loadEventPoliciesSuccess(policies));
+    } catch (e) {
+        console.log(e);
```
add an action for "load event polices faliure"

isabellakutger commented on modules/admin-ui-picard/app/src/thunks/eventDetailsThunks.js:
```diff
@@ -19,6 +21,54 @@ const getHttpHeaders = () => {
+    }
+}
+
+export const fetchHasActiveTransactions = (eventId) => async (dispatch) => {
```
you don't need dispatch here, because you don't make any redux state changes in this method

SimoneSchmidt committed:
Added Requested Changes

Added the changes requested on the pull request "https://github.com/TIK-NFL/opencast/pull/132" on 2021-07-14
